src/pip/_internal/resolution/legacy/resolver.py

```diff
@@ -256,6 +260,34 @@ def _check_skip_installed(self, req_to_install):
         self._set_req_to_reinstall(req_to_install)
         return None
 
+    def _populate_link(self, req):
+        # type: (InstallRequirement) -> None
+        """Ensure that if a link can be found for this, that it is found.
+
+        Note that req.link may still be None - if Upgrade is False and the
+        requirement is already installed.
+
+        If preparer.require_hashes is True, don't use the wheel cache, because
+        cached wheels, always built locally, have different hashes than the
+        files downloaded from the index server and thus throw false hash
+        mismatches. Furthermore, cached wheels at present have undeterministic
+        contents due to file modification times.
```

Comment:
The docstring could use an update.

"upgrade is false" -> requirement may be upgraded, based on upgrade-strategy

Reply:
This would couple this docstring with the implementation of `_is_upgrade_allowed()`. Would it be better to simply mention “if `_is_upgrade_allowed()` is True”?